src/server/server_family.cc

```diff
@@ -1807,6 +1807,25 @@ void ServerFamily::Info(CmdArgList args, ConnectionContext* cntx) {
     append("dispatch_queue_entries", m.facade_stats.conn_stats.dispatch_queue_entries);
   }
 
+  bool has_save_controller = false;
+  size_t save_buffer_bytes = 0;
+  size_t current_snap_keys = 0;
+  size_t total_snap_keys = 0;
+  double perc = 0;
+  {
+    lock_guard lk{save_mu_};
+    if (save_controller_) {
+      has_save_controller = true;
+      save_buffer_bytes = save_controller_->GetSaveBuffersSize();
```

Comment:
I think each of this calls should be inside the relevant if otherwise you invoke the functions even if the user didnt ask for memory/persistence stats

Reply:
but now we need to lock twice.... And we also now in the worst path, do an extra two hops.... The added overhead complexity is adding up...

Reply:
you also already have below a lock for save_mu_ for 
curent_durration_sec = save_controller_->GetCurrentSaveDuration();
I think its better to have them inside the relevant path
You are doing the hops right now regardless of calling the mutex once or twice

Reply:
and also one more comment here. Let's zoom out, from one side, we are saying here that we care about the potential two extra dispatches to the shards and the cost associated with it but at the same time we don't care that we added two extra shard dispatches (for GetCurrentSnapshotProgress) when in fact we did not need them... 

Anyway i made the change -- just thinking out loud :)

Reply:
There is only one hop for GetCurrentSnapshotProgress right? not 2. Or I did not understand what you mean..